tic-tac-toe/src/redux/auth/actions.js

```diff
@@ -0,0 +1,20 @@
+import AuthService from '../../services/LoginService';
+
+export const actions = {
+  VALIDATE_USER: 'VALIDATE_USER',
+  VALIDATE_USER_SUCCESS: 'VALIDATE_USER_SUCCESS',
+  VALIDATE_USER_FAILED: 'VALIDATE_USER_FAILED'
+};
+
+export const actionsCreators = {
+  login: values => async dispatch => {
+    dispatch({ type: actions.VALIDATE_USER });
+    const response = await AuthService.getUserDetail(values);
+    if (response.ok) {
+      dispatch({ type: actions.VALIDATE_USER_SUCCESS, token: response.data, email: values.user });
```

Comment:
We shape actions like `{ type: actions.VALIDATE_USER_SUCCESS, payload:{ token: response.data, email: values.user }}`